### ISSUE TITLE
[Feature] Support snapshot property matchers

### ISSUE THREAD

**Issue by hannoeru** [vitest-dev/vitest#116]
example from jest website:
```js
it('will check the matchers and pass', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
  });
});

// Snapshot
exports[`will check the matchers and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;
```

https://jestjs.io/docs/snapshot-testing#property-matchers
https://jestjs.io/docs/expect#expectanyconstructor

**Comment by sheremet-va** [comment]
We have full snapshot support now 🎉

**Comment by adriencaccia** [comment]
Hey @sheremet-va, it seems that this issue has been closed a bit too soon.

As of v0.0.127, `toMatchSnapshot` and `toMatchInlineSnapshot` are indeed working, but the requested feature of this issue, the property matchers in snapshot, is not implemented.

The example given by @hannoeru is still not working.

**Comment by sheremet-va** [comment]
> Hey @sheremet-va, it seems that this issue has been closed a bit too soon.
> 
> 
> 
> As of v0.0.127, `toMatchSnapshot` and `toMatchInlineSnapshot` are indeed working, but the requested feature of this issue, the property matchers in snapshot tests, is not implemented.
> 
> 
> 
> The example given by @hannoeru is still not working.

But they are implemented. They are accessible on expect object.

**Comment by adriencaccia** [comment]
Unless I'm mistaken, the property matchers are indeed implemented and they are working, what is missing is the extra signatures of `toMatchSnapshot` and `toMatchInlineSnapshot` to support using them in snapshots.

**Comment by sheremet-va** [comment]
Fixed in [v0.0.132](https://github.com/vitest-dev/vitest/releases/tag/v0.0.132) by https://github.com/vitest-dev/vitest/commit/f27d2d6
